File: CS/大二/二上/Algorithm/暑期DS練習題/DS2/java/Pm.java
import java.util.*;
public class Pm {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int ca=1;
        while(sc.hasNextInt()){
             int n=sc.nextInt();
             int limit=sc.nextInt();
             if(n==-1)break;
             int ans=1;
             int copy=n;
             while(n!=1&&n<=limit){
                if(n%2==1){
                    n=n*3+1;
                }
                else{
                    n=n/2;
                }
                ans++;
             }
            if(n>limit)ans--;
            System.out.println("Case "+ca+":"+" K = "+copy+", limit = "+limit+", number of terms = "+ans);
            ca++;
        }
        sc.close();
    }
}

File: CS/大二/二上/Algorithm/暑期DS練習題/DS2/cpp/PM.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    int k,limit,cnt = 1;
    while(cin >> k >> limit){
        if(k == -1 && limit == -1) break;
        int terms = 1;
        int original_k = k;
        while(k != 1){
            
            if(k % 2 == 1){
                k = (3 * k + 1);
            }else{
                k /= 2;
            }

            if(k > limit){
                break;
            }
            terms++;
        }

        printf("Case %d: K = %d, limit = %d, number of terms = %d\n",cnt,original_k,limit,terms);
        cnt++;
    }   
}